api/helpers/game-states/moves/stalemate-request/validate.js

```diff
@@ -36,6 +36,17 @@ module.exports = {
   sync: true,
   fn: ({ currentState, priorStates }, exits) => {
     try {
+
+      // Must not already be considering a stalemate
+      if (currentState.phase === GamePhase.STALEMATE_REQUEST) {
+        throw new Error('game.snackbar.global.alreadyConsideringStalemate');
+      }
+
+      // Must be in main phase
+      if (currentState.phase !== GamePhase.main) {
+        throw new Error('game.snackbar.stalemate.wrongPhase');
+      }
```

Comment:
The bottom error is a superset of the first i.e. would fire every time the phase is "wrong" including when it's a pending stalemate request, but I think explicitly calling out that you can't do that because a stalemate request is already pending is better than the "can't request a stalemate during other moves" catch-all of the second check. 

This was the key change that prevents the bug, which would happen if a stalemate was requested then rejected during non-main phases